src/main/java/games/strategy/engine/framework/startup/login/V1Authenticator.java

```diff
@@ -0,0 +1,125 @@
+package games.strategy.engine.framework.startup.login;
+
+import java.util.Map;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import games.strategy.util.MD5Crypt;
+
+/**
+ * Implements version 1 of the peer-to-peer network game authentication protocol.
+ */
+final class V1Authenticator {
+  /**
+   * Indicates the specified challenge can be processed by this authenticator.
+   *
+   * @param challenge The challenge sent to the client node; must not be {@code null}.
+   *
+   * @return {@code true} if the specified challenge can be processed by this authenticator; otherwise {@code false}.
+   */
+  static boolean canProcessChallenge(final Map<String, String> challenge) {
+    return challenge.containsKey(ChallengePropertyNames.SALT);
+  }
+
+  @VisibleForTesting
+  interface ChallengePropertyNames {
+    String SALT = "Salt";
+  }
+
+  /**
+   * Creates a new challenge for the server node to send to the client node.
+   *
+   * @return The challenge as a collection of properties to be added to the message the server node sends the client
+   *         node; never {@code null}.
+   */
+  static Map<String, String> newChallenge() {
+    return Maps.newHashMap(ImmutableMap.of(
+        ChallengePropertyNames.SALT, MD5Crypt.newSalt()));
+  }
+
+  /**
+   * Indicates the specified response can be processed by this authenticator.
+   *
+   * @param response The response received from the client node; must not be {@code null}.
+   *
+   * @return {@code true} if the specified response can be processed by this authenticator; otherwise {@code false}.
+   */
+  static boolean canProcessResponse(final Map<String, String> response) {
+    return response.containsKey(ResponsePropertyNames.DIGEST);
+  }
+
+  @VisibleForTesting
+  interface ResponsePropertyNames {
+    String DIGEST = "Password";
```

Comment:
I wanted to group the challenge and response property names in different namespaces so I didn't have to include "challenge" or "response" in the constant name itself.  I think it's helpful to distinguish in which context each of these properties is being used to avoid confusion in the future.

Think of these interfaces as cheesy enums for the possible challenge/response property names.  Yeah, having a single enum element looks a bit strange, but you get a bit of pseudo-strong-typing in the sense you know you should only be picking elements from the corresponding enum when crafting your challenge/response.  (And if they were actually API instead of being exposed purely for testing purposes, I probably would make them actual `enum`s.)